### PR TITLE
[Enhancement] Use Streaming Lists when available

### LINKED ISSUE #221: [Enhancement] Use Streaming Lists when available
### Current Behavior
MeshSync performs discovery of specific Kubernetes API types using watches and sharedInformers.

### Desired Behavior
Enhance MeshSync to use [Streaming Lists](https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists) when available in the managed Kubernetes cluster.

### Implementation:
1. Verify whether Streaming Lists are available. If so, use them.

### Acceptance Criteria
1. When Streaming Lists are not available, downgrade to current methodology.
1. Confirm that all of the same resource details are retrieved via Streaming Lists as are retrieved using the current methodology. 

### Contributor Resources

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- See the [Contributing to Meshery.io Website](https://github.com/layer5io/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- See [Meshery site designs](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=110%3A1) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9). 
- Fill-in a [Community Member Form](https://layer5.io/newcomer) and join the [Community Slack](http://slack.meshery.io) for access.

### PR DESCRIPTION
**Pull request by KiptoonKipkurui**
**Description**
This PR Addresses an issue of potential runaway usage of memory due to chucking of reported statuses while watching for Kubernetes resources. Kubernetes 1.27 introduces a WatchList feature still at the alpha stage that allows one to get statuses of objects being watched as a streaming list as opposed to current implementations that uses a list request first before a watch request. This is done by setting `sendInitialEvents=true` param in a watch request. 
This PR fixes #221 

**Notes for Reviewers**
This is still WIP and is allow continuous conversation to allow seamless integration with existing infrastructure

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ x] Yes, I signed my commits.

### CODE REVIEW

File: internal/pipeline/pipeline.go [hunk 1]
```diff
@@ -29,30 +30,48 @@ var (
 	}
 )
 
-func New(log logger.Handler, informer dynamicinformer.DynamicSharedInformerFactory, broker broker.Handler, plConfigs map[string]internalconfig.PipelineConfigs, stopChan chan struct{}) *pipeline.Pipeline {
+func New(log logger.Handler, informer dynamicinformer.DynamicSharedInformerFactory, broker broker.Handler, plConfigs map[string]internalconfig.PipelineConfigs, stopChan chan struct{}, dynamicKube dynamic.Interface) *pipeline.Pipeline {
+	// TODO: best way to check whether WatchList feature is enabled
+	watchList := true
```
Comment (KiptoonKipkurui) [inline internal/pipeline/pipeline.go:35]: I have looked around for ways of determining the enabled features in a Kubernetes cluster, but most knowledge revolves around looking at actual cluster configuration to see what has been enabled and what hasn't been enabled...I would appreciate any pointers to how I can determine programmatically which features have been enabled or not

File: internal/pipeline/step.go [hunk 2]
```diff
@@ -85,3 +93,113 @@ func (si *StartInformers) Cancel() error {
...
+				// when an event is added...
+				case watch.Added:
+					err := w.publishItem(obj.(*unstructured.Unstructured), broker.Add, w.config)
+					if err != nil {
+						w.log.Error(err)
+					}
+					w.log.Info("Received ADD event for: ", obj.(*unstructured.Unstructured).GetName(), "/", obj.(*unstructured.Unstructured).GetNamespace(), " of kind: ", obj.(*unstructured.Unstructured).GroupVersionKind().Kind)
+				// when an event is modified...
+				case watch.Modified:
+					err := w.publishItem(obj.(*unstructured.Unstructured), broker.Update, w.config)
```
Comment (KiptoonKipkurui) [inline internal/pipeline/step.go:166]: For the Update functionality, using informers, one could get the older object and the new object and hence we could compare resource versions to determine whether indeed this object is a valid update or a repeated action. This is not available when using watchers

**Comment by KiptoonKipkurui** [comment]
Using [streaming lists](https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists) boils down to adding the flag `sendInitialEvents=true ` param in a watch request without a List request for the desired Kubernetes object. DynamicInformers also use ListWatch functionality under the hood, and may eventually come to support use of streaming lists.

**Comment by KiptoonKipkurui** [comment]
@leecalcote

**Comment by leecalcote** [comment]
@KiptoonKipkurui, great presentation today, sir.

File: internal/pipeline/pipeline.go (hunk 1 above)
Comment (leecalcote) [reply internal/pipeline/pipeline.go:35]: @KiptoonKipkurui I encourage you to review this PR (its in Rust), but potentially a great reference - http://gitea.lntu.edu.cn/youling/kube/commit/3af6b7f1df216380dd9c9bd13a163fe5281d67fa#diff-d4e0d370315574d0ee0602792eb3abfd2777d3c7

I assume that you have already seen this as well - https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/3157-watch-list/README.md

**Review by MUzairS15 — APPROVED** [review]

**Comment by KiptoonKipkurui** [comment]
@MUzairS15 this isn't ready for a master merge, it will break functionality

File: internal/pipeline/pipeline.go (hunk 1 above)
Comment (KiptoonKipkurui) [reply internal/pipeline/pipeline.go:35]: Thank you for this resources, I will check them out

**Pull request merged**